fiftyone/core/odm/embedded_document.py

```diff
@@ -48,6 +48,9 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.validate()
 
+    def __hash__(self):
```

Comment:
@brimoor : do we need to add unit tests for this?

Reply:
I had to add this in order for the new unit test in this PR to work - needed to convert `list(DynamicEmbeddedDocument)` to `set(DynamicEmbeddedDocument)` - so it's already implicitly tested